api/all/src/main/java/io/opentelemetry/api/trace/TraceFlags.java

```diff
@@ -45,16 +47,21 @@ static TraceFlags getSampled() {
   }
 
   /**
-   * Returns the {@link TraceFlags} converted from the given lowercase hex (base16) representation.
+   * Returns the {@link TraceFlags} converted from the given lowercase hex (base16) representation,
+   * or {@code null} if input is {@code null}, too short, or has invalid characters.
    *
    * @param src the buffer where the hex (base16) representation of the {@link TraceFlags} is.
    * @param srcOffset the offset int buffer.
    * @return the {@link TraceFlags} converted from the given lowercase hex (base16) representation.
-   * @throws NullPointerException if {@code src} is null.
-   * @throws IndexOutOfBoundsException if {@code src} is too short.
-   * @throws IllegalArgumentException if invalid characters in the {@code src}.
    */
+  @Nullable
   static TraceFlags fromHex(CharSequence src, int srcOffset) {
+    // TODO: Avoid calling `subSequence`.
+    if (src == null
+        || src.length() < srcOffset + 2
+        || !OtelEncodingUtils.isValidBase16String(src.subSequence(srcOffset, srcOffset + 2))) {
+      return null;
+    }
```

Comment:
alternative, just for consideration: return TraceFlags.getInvalid(), similar to SpanContext.getInvalid(), callers can check `isValid()`, or they can forget and it will serialize as "00"

Reply:
The 00 flag is actually a valid representation that has a meaningful interpretation

Reply:
Yes I know that

Reply:
Not sure why that's a problem for this proposal?

Reply:
I try to encapsulate somehow the fact that there is an error and leave the caller to decide about default vs not creating the SpanContext

Reply:
just store a `boolean invalid` field in ImmutableTraceFlags

Reply:
caller can check `getInvalid()` if they want to know, same as SpanContext

Reply:
I see... Let me play with that tomorrow since release already happened